hir-type/src/lib.rs

```diff
@@ -67,6 +67,8 @@ pub enum Type {
     Array(Box<Type>, usize),
     /// A tuple of values of the given types
     Tuple(Vec<Type>),
+    /// A dynamically sized list of values of the given type
```

Comment:
Do we know what the concrete representation (in Wasm core terms) this has? This type differs from all the rest in that it has dynamic extent - is this semantically a fat pointer like a slice? Is it a singly-linked cons list (e.g. a pointer to a cons cell, or nil). I'd like to document details like that here for reference.

I'm going to continue with rebasing/merging this PR anyway, but if you could add the docs in one of your other PRs, or upcoming commits, whereever is fine, just want to make sure I know what to do with values of this type within the compiler.